workspace/src/main/kotlin/com/cosmotech/workspace/service/WorkspaceServiceImpl.kt

```diff
@@ -306,7 +307,8 @@ internal class WorkspaceServiceImpl(
   ): Workspace {
     organizationService.getVerifiedOrganization(organizationId)
     sendAddWorkspaceToDatasetEvent(organizationId, datasetId, workspaceId)
-    return addDatasetToLinkedDatasetIdList(organizationId, workspaceId, datasetId)
+    return checkReadSecurity(
```

Comment:
Here the `checkReadSecutiry` is not relevant **BUT** there is something missing: the check that the current user has permissions to modified the workspace... this is another bug but can you change that also in your PR (same thing with `unlinkDataset`)